src/libraries/System.Private.CoreLib/src/System/Globalization/Normalization.cs

```diff
@@ -35,5 +56,63 @@ internal static string Normalize(string strInput, NormalizationForm normalizatio
                 NlsNormalize(strInput, normalizationForm) :
                 IcuNormalize(strInput, normalizationForm);
         }
+
+        internal static bool TryNormalize(ReadOnlySpan<char> source, Span<char> destination, out int charsWritten, NormalizationForm normalizationForm = NormalizationForm.FormC)
+        {
+            CheckNormalizationForm(normalizationForm);
+
+            if (GlobalizationMode.Invariant || source.IsEmpty)
+            {
+                // In Invariant mode we assume all characters are normalized.
+                // This is because we don't support any linguistic operation on the strings
+                charsWritten = 0;
+                return true;
+            }
+
+            if (Ascii.IsValid(source))
+            {
+                if (destination.Length < source.Length)
+                {
+                    charsWritten = 0;
+                    return false;
+                }
+
+                source.CopyTo(destination);
+                charsWritten = source.Length;
+                return true;
+            }
+
+            return GlobalizationMode.UseNls ?
+                NlsTryNormalize(source, destination, out charsWritten, normalizationForm) :
+                IcuTryNormalize(source, destination, out charsWritten, normalizationForm);
+        }
+
+        internal static int GetNormalizedLength(this ReadOnlySpan<char> source, NormalizationForm normalizationForm = NormalizationForm.FormC)
+        {
+            CheckNormalizationForm(normalizationForm);
+
+            if (GlobalizationMode.Invariant || source.IsEmpty || Ascii.IsValid(source))
+            {
+                // In Invariant mode we assume all characters are normalized.
+                // This is because we don't support any linguistic operation on the strings
+                return source.Length;
```

Comment:
The previously commented on charsWritten = 0 also doesn't match with this.

Reply:
I fixed the other place.